frame/balances/src/lib.rs

```diff
@@ -946,39 +946,101 @@ impl<T: Config<I>, I: Instance> Currency<T::AccountId> for Module<T, I> where
+		let mut maybe_from_acc_call_post_mutation = false;
+		let mut maybe_to_acc_call_post_mutation = false;
+		let mut from_acc: AccountData<T::Balance> = Default::default();
+		let mut to_acc: AccountData<T::Balance> = Default::default();
+
+		T::AccountStore::try_mutate_exists(dest, |maybe_to_acc| -> DispatchResult {
+			let to_acc_is_new = maybe_to_acc.is_none();
+			to_acc = maybe_to_acc.take().unwrap_or_default();
+
+			T::AccountStore::try_mutate_exists(transactor, |maybe_from_acc| -> DispatchResult {
+				let from_acc_is_new = maybe_from_acc.is_none();
+				from_acc = maybe_from_acc
+					.take()
+					.unwrap_or_default();
+
+				from_acc.free = from_acc.free.checked_sub(&value)
 					.ok_or(Error::<T, I>::InsufficientBalance)?;
 
-				// NOTE: total stake being stored in the same type means that this could never overflow
+				// NOTE: total stake being stored in the same type means
+				// that this could never overflow
 				// but better to be safe than sorry.
-				to_account.free = to_account.free.checked_add(&value).ok_or(Error::<T, I>::Overflow)?;
+				to_acc.free = to_acc
+					.free.checked_add(&value)
+					.ok_or(Error::<T, I>::Overflow)?;
 
 				let ed = T::ExistentialDeposit::get();
-				ensure!(to_account.total() >= ed, Error::<T, I>::ExistentialDeposit);
+				ensure!(
+					to_acc.total() >= ed,
+					Error::<T, I>::ExistentialDeposit,
+				);
 
 				Self::ensure_can_withdraw(
 					transactor,
 					value,
 					WithdrawReasons::TRANSFER,
-					from_account.free,
+					from_acc.free,
 				)?;
 
 				let allow_death = existence_requirement == ExistenceRequirement::AllowDeath;
 				let allow_death = allow_death && system::Module::<T>::allow_death(transactor);
-				ensure!(allow_death || from_account.free >= ed, Error::<T, I>::KeepAlive);
+				ensure!(allow_death || from_acc.free >= ed, Error::<T, I>::KeepAlive);
+
+				maybe_from_acc_call_post_mutation =
+					from_acc.total() < T::ExistentialDeposit::get();
+				if from_acc.total() < T::ExistentialDeposit::get() {
+					*maybe_from_acc = None;
+				} else {
+					*maybe_from_acc = Some(from_acc.clone());
+				}
+
+				maybe_to_acc_call_post_mutation =
+					to_acc.total() < T::ExistentialDeposit::get();
+				if to_acc.total() < T::ExistentialDeposit::get() {
+					*maybe_to_acc = None;
+				} else {
+					*maybe_to_acc = Some(to_acc.clone());
+				}
+
+				let maybe_from_acc_endowed = if from_acc_is_new { Some(from_acc.free) } else { None };
+				let maybe_to_acc_endowed = if to_acc_is_new { Some(to_acc.free) } else { None };
 
+				if let Some(from_acc_endowed) = maybe_from_acc_endowed {
+					Self::deposit_event(
+						RawEvent::Endowed(transactor.clone(), from_acc_endowed)
+					);
+				}
+
+				if let Some(to_acc_endowed) = maybe_to_acc_endowed {
+					Self::deposit_event(
+						RawEvent::Endowed(dest.clone(), to_acc_endowed)
+					);
+				}
 				Ok(())
 			})
 		})?;
 
+		if maybe_from_acc_call_post_mutation {
+			let _ = Self::post_mutation(transactor, from_acc);
```

Comment:
I don't think `let _ =` is needed. It is always a code smell to ignore thing without any explanation

Reply:
Sure will ignore it.